components/brave_wallet/browser/brave_wallet_provider_impl.cc

```diff
@@ -13,35 +13,19 @@
 namespace brave_wallet {
 
 BraveWalletProviderImpl::BraveWalletProviderImpl(
-    EthJsonRpcController* rpc_controller,
+    mojo::PendingRemote<mojom::EthJsonRpcController> rpc_controller,
```

Comment:
we can access the EthJsonRpcController directly instead of going through `BraveWalletProviderImpl`. Same thing in `WalletHandler`

Reply:
Posted here to track this separately: https://github.com/brave/brave-browser/issues/17184